### PR TITLE
doc: Document traversal interfaces and classes

### PR DESCRIPTION
**Pull request by eviltak**
Adds missing documentation to classes and interfaces in the `org.terasology.blockNetwork.traversal` package.

## Documents
- `TraversalResult` and its fields
- `BreadthFirstTraversal` methods
- `BreadthFirstTraversalWithPath` methods

### CODE REVIEW

File: src/main/java/org/terasology/blockNetwork/traversal/TraversalResult.java [hunk 1]
```diff
@@ -18,10 +18,33 @@
...
+ * The result of a node traversal performed on a graph. It contains a flag to stop the traversal, a predicate for selecting consequent 
+ * nodes to be traversed, the returned result of the traversal, and the value propagated through the traversal.
+ * 
+ * @param <T> Type of nodes in the graph.
+ * @param <U> Return value type from the traversal.
+ * @param <V> Value passed from node visited into nodes that are outgoing from that node.
+ */
 public final class TraversalResult<T, U, V> {
+    /**
+     * A flag specifying whether to stop the traversal or not.
```
Comment (pollend) [inline src/main/java/org/terasology/blockNetwork/traversal/TraversalResult.java]: can you change these to single comment lines //

File: src/main/java/org/terasology/blockNetwork/traversal/BreadthFirstTraversal.java [hunk 2]
```diff
@@ -24,5 +24,13 @@
  * @param <V> Value passed from node visited into nodes that are outgoing from that node.
  */
 public interface BreadthFirstTraversal<T, U, V> {
+    /**
+     * Traverses the given node in the graph.
+     * This method is invoked for each node in a traversal as directed by the traversal definition and the traversal result.
```
Comment (vampcat) [inline src/main/java/org/terasology/blockNetwork/traversal/BreadthFirstTraversal.java]: How does the "traversal result" direct the order of node traversal? If anything, i would have assumed the opposite - that the traversal dictates the result.

File: src/main/java/org/terasology/blockNetwork/traversal/BreadthFirstTraversal.java [hunk 3]
```diff
@@ -24,5 +24,13 @@
...
+     * 
+     * @param node The type of nodes in the graph.
```
Comment (vampcat) [inline src/main/java/org/terasology/blockNetwork/traversal/BreadthFirstTraversal.java]: I'm not sure what you are doing here.

`node` is *not* the type of nodes in the graph, it is the node that is currently being traversed.

I get the feeling that you've essentially blindly copied the class level javadocs on this class, without even reading them once. I seriously did not expect that from you.

File: src/main/java/org/terasology/blockNetwork/traversal/BreadthFirstTraversalWithPath.java [hunk 4]
```diff
@@ -27,5 +27,14 @@
  * @param <V> Value passed from node visited into nodes that are outgoing from that node.
  */
 public interface BreadthFirstTraversalWithPath<T, U, V> {
+    /**
+     * Traverses the given node in the graph.
+     * This method is invoked for each node in a traversal as directed by the traversal definition and the traversal result.
```
Comment (vampcat) [inline src/main/java/org/terasology/blockNetwork/traversal/BreadthFirstTraversalWithPath.java]: Same comment as above

File: src/main/java/org/terasology/blockNetwork/traversal/BreadthFirstTraversalWithPath.java [hunk 5]
```diff
@@ -27,5 +27,14 @@
...
+     * 
+     * @param node The type of nodes in the graph.
```
Comment (vampcat) [inline src/main/java/org/terasology/blockNetwork/traversal/BreadthFirstTraversalWithPath.java]: Same comment as above

File: src/main/java/org/terasology/blockNetwork/traversal/TraversalResult.java (hunk 1 above)
Comment (vampcat) [reply src/main/java/org/terasology/blockNetwork/traversal/TraversalResult.java]: Agree with Michael on this one, having simple comments seems better to me.

File: src/main/java/org/terasology/blockNetwork/traversal/BreadthFirstTraversal.java (hunk 3 above)
Comment (vampcat) [reply src/main/java/org/terasology/blockNetwork/traversal/BreadthFirstTraversal.java]: In response to an earlier, unedited version of this comment: I'm sorry if it seemed like I was being rude, I definitely did not want it to come out that way.

File: src/main/java/org/terasology/blockNetwork/traversal/BreadthFirstTraversal.java (hunk 2 above)
Comment (eviltak) [reply src/main/java/org/terasology/blockNetwork/traversal/BreadthFirstTraversal.java]: The result influences the order through the predicate which specifies which node to traverse next and which not to.

File: src/main/java/org/terasology/blockNetwork/traversal/BreadthFirstTraversal.java (hunk 3 above)
Comment (eviltak) [reply src/main/java/org/terasology/blockNetwork/traversal/BreadthFirstTraversal.java]: You don't have to be. You were completely right, and this kind of sloppiness should not be expected. I apologize; once I have access to my work environment in a few hours, I can guarantee you that my work will return to the level it previously was. I shall take a good look at everything again.

**Pull request merged**